File: challenges/3-tipos-de-triangulo/java/felipersteles/triangulo.java

import java.util.*;
import java.util.Arrays;

class TiposDeTriangulo {
    public static void main(String[ ] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Digite tres valores: ");
        double A = scanner.nextInt();
        double B = scanner.nextInt();
        double C = scanner.nextInt();
        scanner.close();

        Triangulo( A,  B,  C);

    }

    private static void Triangulo(double A, double B, double C){

        System.out.println();

        double[] aux = {A,B,C};
        Arrays.sort(aux);

        A = aux[2];
        B = aux[1];
        C = aux[0];

        if(A >= (B+C)){
            System.out.println("NAO FORMA TRIANGULO");
        } 
        else{
            if( (A*A) == ( (B*B) + (C*C) ) ){
                System.out.println("TRIANGULO RETANGULO");
            }
            if( (A*A) > ( (B*B) + (C*C) ) ){
                System.out.println("TRIANGULO OBTUSANGULO");
            }
            if( (A*A) < ( (B*B) + (C*C) ) ){
                System.out.println("TRIANGULO ACUTANGULO");
            }
            if (A == B && B == C){
                System.out.println("TRIANGULO EQUILATERO");
            }
            if (A == B && A != C){
                System.out.println("TRIANGULO ISOSCELES");
            }
            if (B == C && B != A){
                System.out.println("TRIANGULO ISOSCELES");
            }
            if (A == C && A != B){
                System.out.println("TRIANGULO ISOSCELES");
            }
        }
    }
}

File: challenges/3-tipos-de-triangulo/php/kayehol/triangulo.php
<?php
// gerencia input
$values = [];
forEach($argv as $key => $arg) {
    if ($key !== 0 && is_numeric($arg)) 
        array_push($values, number_format(floatval($arg), 1));
}
if (count($values) != 3) {
    echo "[Argumentos inválidos]: Insira 3 números".PHP_EOL;
    die();
}
echo "Entrada: ".implode(" ",$values).PHP_EOL;

// processa
rsort($values, SORT_NUMERIC);
$numsUnicos = array_unique($values, SORT_NUMERIC);
$a = $values[0];
$b = $values[1];
$c = $values[2];

if ($a >= ($b + $c)) { 
    echo "NAO FORMA TRIANGULO".PHP_EOL;
    die();
}

$sqA = $a * $a;
$sqB = $b * $b;
$sqC = $c * $c;
$sqMidLow = $sqB + $sqC;

if (count($numsUnicos) == 1) {
    $ladoTriangulo = 'TRIANGULO EQUILATERO';
} else if (count($numsUnicos) == 2) {
    $ladoTriangulo = 'TRIANGULO ISOSCELES';
}

if ($sqA == $sqMidLow) {
    $tipoTriangulo = "TRIANGULO RETANGULO";
} elseif ($sqA > $sqMidLow) {
    $tipoTriangulo = "TRIANGULO OBTUSANGULO";
} else {
    $tipoTriangulo = "TRIANGULO ACUTANGULO";
}

// exibe resultado
echo $tipoTriangulo.PHP_EOL;
echo $ladoTriangulo.PHP_EOL;